### DISCUSSION TITLE
Livewire listener and refresh

### DISCUSSION THREAD

**Discussion by Seiliaz** [livewire/livewire#6648] [1. Help]
Hello, everyone!
in the latest update of livewire we saw that the $listeners variable was replaced with $[On()] attribute. and the $refresh didn't change because of it's popularity (in the livewire blade file)
but here is my question; I used to do this to refresh a component when a specific message was emitted:
$listeners = ["..." => '$refresh']
so how can I do this without the $listeners and with the attributes?

**Comment by reasecret** [comment]
I'm using on attribute with render method:

```
#[On('refreshPage')]
public function render () {
//
}
```

**Comment by krekas** [comment]
```php
#[On('refresh-list')]
public function refresh() {}
```

**Reply by whousehomestay** [reply]
Yeah, actually we can make any-named function, and livewire will always trigger the render method and refresh the component...

**Reply by levu42** [reply]
I think I have a feature request: Is it possible to extend the `#[On]` attribute to be also registered on a component?

```
#[On('username-changed')]
class UserDropdown extends Component
{
}
```

What do you think, @calebporzio? If you like the idea, I'd try to create a PR for it.
